File: 백준/Bronze/1978. 소수 찾기/소수 찾기.java
import java.util.*;

public class Main {
	public static void main(String[] args) {
		List<Boolean> primeList = new ArrayList<>();
		primeList.add(false); 	//0
		primeList.add(false);	//1
				
		for (int i = 2; i < 1000 + 1; i++) {
				primeList.add(i, true);
		}
				
		for (int i = 2; i <= Math.sqrt(1000); i++) {
				if(primeList.get(i)) {
					for (int j = i * i; j <= 1000 ; j += i) {
						primeList.set(j, false);
					}
				}
		}
		
		Scanner scanner = new Scanner(System.in);
		int N = scanner.nextInt();
		int cnt = 0;
		for (int i = 0; i < N; i++) {
			int x = scanner.nextInt();
			if(primeList.get(x)) cnt++;
		}
		System.out.println(cnt);
	}
}

File: 백준/Bronze/1978. 소수 찾기/소수 찾기.py
import math
import sys

def main():
    prime_list = [False, False] + [True] * 1000

    for i in range(2, int(math.sqrt(1000)) + 1):
        if prime_list[i]:
            for j in range(i * i, 1001, i):
                prime_list[j] = False

    N = int(sys.stdin.readline().strip())
    numbers = list(map(int, sys.stdin.readline().split()))
    cnt = 0
    for x in numbers:
        if prime_list[x]:
            cnt += 1

    print(cnt)

if __name__ == "__main__":
    main()
